### ISSUE TITLE
Aborting cosmic-ray init leaves a partial config file lying around

### ISSUE THREAD

**Issue by rob-smallshire** [sixty-north/cosmic-ray#274]
Consider altering cosmic-ray init so no changes are made to the filesystem until it is in a position to succeed.

**Comment by rob-smallshire** [comment]
Subsequent invocations of the same command cause cosmic-ray init to fail with this error, until the partial config file is manually removed.

```
$ cosmic-ray init cosmic-ray-pytest-local.conf session1
Traceback (most recent call last):
  File "/Users/rjs/.virtualenvs/asm68/bin/cosmic-ray", line 11, in <module>
    sys.exit(main())
  File "/Users/rjs/.virtualenvs/asm68/lib/python3.6/site-packages/cosmic_ray/cli.py", line 298, in main
    common_option_handler=common_option_handler)
  File "/Users/rjs/.virtualenvs/asm68/lib/python3.6/site-packages/docopt_subcommands/__init__.py", line 69, in main
    result = commands(argv)
  File "/Users/rjs/.virtualenvs/asm68/lib/python3.6/site-packages/docopt_subcommands/subcommands.py", line 134, in __call__
    return handler(config)
  File "/Users/rjs/.virtualenvs/asm68/lib/python3.6/site-packages/cosmic_ray/cli.py", line 100, in handle_init
    if 'timeout' in config:
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by abingham** [comment]
Thanks! This should be fixed in aa7a04d458d55d8b182418fc2e2b9dd1fcd0adb9. We fully construct the config before opening the file handle. Reopen if this doesn't work for you.
